test/e2e/tests/http3.go

```diff
@@ -0,0 +1,220 @@
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTP3Test)
+}
+
+var HTTP3Test = suite.ConformanceTest{
+	ShortName:   "Http3",
+	Description: "Testing http3 request",
+	Manifests:   []string{"testdata/http3.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("Send http3 request", func(t *testing.T) {
+			namespace := "gateway-conformance-https"
+			routeNN := types.NamespacedName{Name: "http3-route", Namespace: namespace}
+			gwNN := types.NamespacedName{Name: "http3-gateway", Namespace: namespace}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			server := "www.example.com"
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/get",
+					Headers: map[string]string{
+						"User-Agent":        "quic-go HTTP/3",
+						"X-Envoy-Internal":  "true",
+						"X-Forwarded-Proto": "https",
+					},
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: namespace,
+			}
+
+			req := http.MakeRequest(t, &expectedResponse, server+":443", "HTTPS", "https")
+			req.CertPem = []byte("-----BEGIN CERTIFICATE-----\nMIIDOzCCAiOgAwIBAgIUZTLKDkhVrxfgt9megu5uiSRwRxswDQYJKoZIhvcNAQEL\nBQAwLTEVMBMGA1UECgwMZXhhbXBsZSBJbmMuMRQwEgYDVQQDDAtleGFtcGxlLmNv\nbTAeFw0yNDAzMDIxNDU3MTJaFw0yNTAzMDIxNDU3MTJaMC0xFTATBgNVBAoMDGV4\nYW1wbGUgSW5jLjEUMBIGA1UEAwwLZXhhbXBsZS5jb20wggEiMA0GCSqGSIb3DQEB\nAQUAA4IBDwAwggEKAoIBAQDJBSwf0XJkglooQbzQ6Io/M7gwmbhTjpQPX7P2/ZV6\ncdFsXBUF0X91wABtCtibv+x4if+yvPhHzBERzhjwwQitKZiewhOFoSz5ZyKT8HXd\n+Y06iRzWEnGi2i/98YiBFsG1xc5mHTLgyi6PjJDzGVdsNrL7pE8aM3R9sGrkG4PZ\n5ZWFlpbxX9eUz9dplLfLX7jKETbyKsiwcHihXAY4mdpuaUVifz35tSpnQ3P/RoNw\nNl7/gvRKqSK1a5ByYqVANV/c+O1R9MaR8kG9Cmj2PZPZm4vW/bJIs2R7z6ygGiDx\n4GyMT4MA52pk/dDkV2EtvFk5JFXoMjNPcNRGjuusRe17AgMBAAGjUzBRMB0GA1Ud\nDgQWBBQXeg9HF7TIRHUWOsF5dd0m68zbxTAfBgNVHSMEGDAWgBQXeg9HF7TIRHUW\nOsF5dd0m68zbxTAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCS\nO7hTBUp3MjsOBGXmuuDPatz0eR05Dq4O6Etsn9KVRND0o8iULqYGPMsozkueRc0y\n/Ra1aTbNXm0n5gwNBgJeSOIOfLN0r0L5uBFSMjTPAo6qPkDzK5gXHfzpoxgDYCEy\nSuzYOIHliWoG6K2ldfMx7psMe3ZiR9SA5evOv3VKJDrrhO57niRmDhZKsADWDUFH\nyyximSiPKjFJLJ+7B4N+7TAmxDjMd2vre7qfRL/AFTc7zIkQBG+JoXbusOw1yGcm\n8IV50ZroyupxND625FgawISPYUelwS39x4jA7QNH0Tzgzp+Ao0N8Ck9H4waNcJt8\njk27Qn0mC7RsIqb981eH\n-----END CERTIFICATE-----")
+			req.Server = server
+			cReq, cResp, err := CaptureRoundTrip(req, gwAddr)
+			if err != nil {
+				t.Errorf("failed to get expected response: %v", err)
+			}
+
+			if err := http.CompareRequest(t, &req, cReq, cResp, expectedResponse); err != nil {
+				t.Errorf("failed to compare request and response: %v", err)
+			}
+		})
+	},
+}
+
+func CaptureRoundTrip(request roundtripper.Request, gwAddr string) (*roundtripper.CapturedRequest, *roundtripper.CapturedResponse, error) {
```

Comment:
should we equip this with retry ability?

Reply:
why? It seems unnecessary and there is no such ability in `sigs.k8s.io/gateway-api` either